persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/StargateQueryHandler.java

```diff
@@ -217,7 +219,8 @@ protected void authorizeByToken(Map<String, ByteBuffer> customPayload, CQLStatem
             authenticationSubject,
             castStatement.keyspace(),
             castStatement.columnFamily(),
-            SourceAPI.CQL);
+            SourceAPI.CQL,
+            EMPTY_HEADERS);
```

Comment:
Same concern as in https://github.com/stargate/stargate/pull/581/files#r559749974

Reply:
For stargate query handlers we should pass EMPTY_HEADERS. Only the CNDB StargasteQueryHandler will extract the tenant id from ClientStateWithTenant and pass it to the components. It was discussed on slack.